lms/djangoapps/mobile_api/badges/urls.py

```diff
@@ -0,0 +1,37 @@
+"""
+URLs for badges API
+"""
+from django.conf.urls import patterns, url
+from django.conf import settings
+
+from mobile_api.badges.views import BadgeClassDetail, UserBadgeAssertions
+from openedx.core.djangoapps.user_api.urls import USERNAME_PATTERN
+
+BADGE_PATTERN = r'(?P<issuing_component>[-\w]*)/(?P<slug>[-\w]+)'
+
+BADGE_PATTERN_COURSE = BADGE_PATTERN + '/course/' + settings.COURSE_ID_PATTERN
+
+BADGE_PATTERN_WILDCARD = BADGE_PATTERN + '/course/(?P<course_id>[*])'
+
+urlpatterns = patterns(
```

Comment:
It became apparent during implementation that if Badge Classes could not be scoped to courses more readily, we'd end up in a heap of trouble. So there are some badge classes that require a course to be specified in order to look them up.